### DISCUSSION TITLE
Refreshing Bootstrap Servers for AdminClient on Runtime

### DISCUSSION THREAD

**Discussion by theurichde** [provectus/kafka-ui#3516] [Q&A] [answered]
Hello!

I am wondering if there is any way to reconnect to a Kafka cluster (with changed IPs) without having to restart the whole application?

We have a staging cluster running, which is teared down outside our business hours. The cluster runs in the cloud and the brokers get new IPs assigned after spinning the cluster up again. As bootstrap servers is a DNS record (kafka-xxxx.awesome.cloud:4123) set. Kafka UI then isn't able to reconnect to those brokers, as the AdminClient fetches the broker information only on application startup.

I would like to avoid restarting the application every time the staging cluster is spin up again. 
I couldn't find a suitable API endpoint for that use case :-/

Do you have any ideas how to achieve that?
I already switched `client.dns.lookup` to `resolve_canonical_bootstrap_servers_only`, but that does also only apply on application start.

Thanks!

**Comment by Haarolean** [comment]
Hey, sorry for the delay.

Out of the box we have [configuration wizard](https://docs.kafka-ui.provectus.io/configuration/configuration-wizard).
That might help, you'd just need to edit the cluster and reapply the config right after.

Meanwhile, let me take a look at other possible solutions.

**Comment by iliax** [comment, marked as answer]
Hello @theurichde 
I tested the behavior you described - and it works fine for me. 

How I tested that: 
1. run kafka-ui & kafka broker in the same docker network
2. disconnected broker and connected back with new ip
3. restarted broker container to reenable ports listening

After all that steps kafka-ui is able to connect to the cluster. 

Can you please provide some additional details and logs that you got?
